src/main/java/cloud/eppo/configuration/ConfigurationBuffer.java

```diff
@@ -0,0 +1,116 @@
+package cloud.eppo.configuration;
+
+import cloud.eppo.ufc.dto.*;
+import cloud.eppo.ufc.dto.adapters.EppoModule;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Wrapper object passed between the various Configuration handling classes encapsulating flag and
+ * bandit configuration as well as whether the config is obfuscated.
+ */
+public class ConfigurationBuffer {
+
+  private static final Logger log = LoggerFactory.getLogger(ConfigurationBuffer.class);
+
+  private final Map<String, BanditReference> banditReferences;
+  private final Map<String, FlagConfig> flags;
+  private Map<String, BanditParameters> bandits;
```

Comment:
minor: is it hard to make `bandits` final as well? I think it's best to make this object completely immutable, so there's less chance for inconsistencies and synchronization issues

Reply:
it was.. and also wasn't. Moving to a Builder adds a few lines but makes building immutable Configuration objects pretty clean.

Since we want to conditionally load bandits, we need to parse the flag config and compare against some set of bandit models. Moving that workload into the `Configuration.Builder` now eliminates another race (previously, the flag config was set into the config store, which parsed and provided the bandit models needed to be loaded, thus potentially having an invalid config until the bandit response was set).